apps/assets/migrations/0015_automationexecution_type.py

```diff
@@ -0,0 +1,28 @@
+# Generated by Django 4.1.13 on 2025-03-13 09:14
+
+from django.db import migrations, models
+
+
+def migrate_execution_type(apps, schema_editor):
+    execution_model = apps.get_model('assets', 'AutomationExecution')
+    execution_objs = []
+    for execution in execution_model.objects.all():
+        snapshot = execution.snapshot
+        execution.type = snapshot.get('type', '')
+        execution_objs.append(execution)
+    execution_model.objects.bulk_update(execution_objs, ['type'])
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ('assets', '0014_alter_automationexecution_duration'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='automationexecution',
+            name='type',
+            field=models.CharField(default='', max_length=16, verbose_name='Type'),
+        ),
+        migrations.RunPython(migrate_execution_type)
+    ]
```

Comment:
Here are some observations about this file:

1. There's no explicit error handling (no try/except), so we should add that if there could be issues when running the migration.

2. We can improve the `migrate_execution_type` function using SQLAlchemy to update objects instead of iterating through all rows, which is more efficient.

3. The database operations have been done before adding any data into the table because it seems like they might not get executed correctly without an initial insert first.

4. If you find other changes over time, also keep these considerations in mind.